### PR TITLE
PV E2E:  dynamically provisioned volume should not create in unmanaged zone

### PR DESCRIPTION
**Pull request by copejon**
**What this PR does / why we need it**:
Adds e2e test to that attempts to provision a volume in an unmanaged zone and fails on success.  This is to catch regressions of #31948.

cc @jeffvance 
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/volume_provisioning.go [hunk 1]
```diff
@@ -121,6 +123,61 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
...
+
+			gceCloud, err := getGCECloud()
+			Expect(err).NotTo(HaveOccurred())
+
+			// Get the master node zone
+			managedZones, err = gceCloud.GetAllZones()
+			Expect(err).NotTo(HaveOccurred())
+
+			// Get a list of all zones in the project
+			compSvc := gceCloud.GetComputeService()
```
Comment (jeffvance) [inline test/e2e/volume_provisioning.go]: Is there a helper func to get all zones and/or get a specific zone?

File: test/e2e/volume_provisioning.go [hunk 2]
```diff
@@ -121,6 +123,61 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
...
+			call := compSvc.Zones.List(framework.TestContext.CloudConfig.ProjectID)
```
Comment (jeffvance) [inline test/e2e/volume_provisioning.go]: Is there a helper func to get all zones and/or get a specific zone?

File: test/e2e/volume_provisioning.go [hunk 3]
```diff
@@ -26,9 +26,9 @@ import (
 	storageutil "k8s.io/kubernetes/pkg/apis/storage/v1beta1/util"
 	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
 	"k8s.io/kubernetes/test/e2e/framework"
-
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/util/sets"
```
Comment (jsafrane) [inline test/e2e/volume_provisioning.go]: this belongs to the chunk of k8s imports above.

File: test/e2e/volume_provisioning.go [hunk 4]
```diff
@@ -121,6 +121,56 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
...
+				allZones.Insert(z.Name)
+			}
+
+			// Get the subset of zones not managed by k8s
+			var unmanagedZone string
+			var popped bool
+			unmanagedZones := allZones.Difference(managedZones)
+			// And select one of them at random.
+			if unmanagedZone, popped = unmanagedZones.PopAny(); ! popped {
+				framework.Failf("No unmanaged zones found.")
```
Comment (jsafrane) [inline test/e2e/volume_provisioning.go]: I'd skip the test in this case - it's not a failure.

File: test/e2e/volume_provisioning.go [hunk 5]
```diff
@@ -121,6 +121,56 @@ var _ = framework.KubeDescribe("Dynamic provisioning", func() {
...
+			}
+
+			By("Creating a StorageClass for the unmanaged zone")
+			sc := newStorageClass()
+			// Set an unmanaged zone.
+			sc.Parameters = map[string]string{"zone": unmanagedZone,}
+			sc, err = c.Storage().StorageClasses().Create(sc)
+			defer Ω(c.Storage().StorageClasses().Delete(sc.Name, nil)).Should(Succeed())
```
Comment (jsafrane) [inline test/e2e/volume_provisioning.go]: Please use `Expect` syntax instead of `Ω`.

**Review by jsafrane — CHANGES_REQUESTED** [review]

**Comment by jsafrane** [comment]
@k8s-bot ok to test

**Comment by jsafrane** [comment]
```
I0208 04:53:19.259] Verifying hack/make-rules/../../hack/verify-gofmt.sh
I0208 04:53:28.013] !!! 'gofmt -s -w' needs to be run on the following files: 
I0208 04:53:28.013] ./test/e2e/volume_provisioning.go
I0208 04:53:28.022] FAILED   hack/make-rules/../../hack/verify-gofmt.sh	9s
```

**Comment by jsafrane** [comment]
now it looks fine, @copejon please squash the commits and I'll lgtm it (is it a verb?)

**Review by jsafrane — APPROVED** [review]

**Comment by jsafrane** [comment]
@copejon, hack/verify-gofmt.sh is still not happy, please reformat test/e2e/volume_provisioning.go

**Comment by jsafrane** [comment]
@k8s-bot gce etcd3 e2e test this
@k8s-bot kops aws e2e test this

**Pull request merged**